### PR TITLE
Rdoc

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Enhancements for open, parse, minor tweaks.

### CODE REVIEW

**Comment by BurdetteLamar** [comment]
@kou, did not mean to merge this.  What to do now?

File: lib/csv.rb [hunk 1]
```diff
@@ -864,18 +863,18 @@ def filter(input=nil, output=nil, **options)
     #   ["baz", "2"]
     #
     # Read rows from an \IO object:
-    #   CSV.foreach(io) {|row| p row } # => 21
+    #   CSV.foreach(File.open(path)) {|row| p row } # => 21
```
Comment (kou) [inline lib/csv.rb:866]: We should not use no `close` `File.open` in example. It's a bad mannar:

```ruby
File.open(path) do |input|
  CSV.foreach(io) {row p row } # => 21
end
```

File: lib/csv.rb [hunk 2]
```diff
@@ -1022,10 +1021,10 @@ def generate_line(row, **options)
 
     #
     # :call-seq:
-    #   open( filename, mode = "rb", **options ) { |faster_csv| ... }
-    #   open( filename, **options ) { |faster_csv| ... }
-    #   open( filename, mode = "rb", **options )
-    #   open( filename, **options )
+    #   open(file_path, mode = "rb", **options ) -> new_csv
+    #   open(io, mode = "rb", **options ) -> new_csv
+    #   open(file_path, mode = "rb", **options ) { |csv| ... } -> new_csv
```
Comment (kou) [inline lib/csv.rb:1026]: It returns the block's return value not the yielded `csv` instance.

**Comment by BurdetteLamar** [comment]
Ok, thanks.  Will study tomorrow.

**Pull request merged**